### DISCUSSION TITLE
What accessibility issues you have encountered with Bitcoin?

### DISCUSSION THREAD

**Discussion by GBKS** [BitcoinDesign/Guide#449] [General]
We had a conversation around accessibility in [yesterdays' jam session](https://github.com/BitcoinDesign/Meta/issues/118), and one of the questions was around what the biggest problems are at the moment. Are they of technical nature, like lack of support for screen readers? Are they cultural and language barriers? Is it a lack of documentation & education? Let's try to gather and discuss information & research here to narrow in on what the problems are and how to tackle them.

**Comment by jason-me** [comment]
I would say all of the above at this point. Most of the UIs and user documentation are pretty inaccessible and I don't know of a single accessible hardware wallet that someone with a visual impairment can easily use. Icy Wallet was attempting this but that project hasn't been updated in a while. I do know Christopher Allen from Bitcoin Commons was trying to implement fully accessible features in Gordion walllet a while back, but am not sure where he left off and haven't had a chance to test it with a screen reader since. Good news is bitcoin is text based so it has that going for it in Core. All of the great new tools, user experiences, and resources though are largely inaccessible from what I have seen.

**Reply by GBKS** [reply]
Thanks for chiming in. I hope we can improve on this. What do you think is the most urgent thing to work on?

There are two things happening on the topic in the community. Not much, but it's something:
- Ensuring the website is accessible ([PR](https://github.com/BitcoinDesign/Guide/pull/770))
- Creating a basic page about accessibility in the design guide ([issue](https://github.com/BitcoinDesign/Guide/issues/754))

I hope we can continue to systematically build up more information and resources on this topic, but we are missing people with experience who can push this topic regularly.

There's also a [new GUI](https://github.com/BitcoinDesign/Bitcoin-Core-App) for Bitcoin Core in the works and I hope we can consider accessibility from the start. Out of anything, this is the wallet that should be the most universally accessible, as it is intended as a reference implementation.

**Comment by jason-me** [comment]
I agree 100%. I have a lot of experience in this area and would be glad to help. It is hard to prioritize without doing a full audit, but if I had to guess I would say.

1) Enabling users with visual disabilities to more easily take control of and store their private keys (Maybe a guide that points to the most accessible wallets and resources available today)
2) Documentation and website conformance to WCAG 2.1 AA guidelines for accessibility (Audit and remediation of existing front end a11y issues)
3) User Design Guide a11y resources for wallet developers (i.e. Interaction Design Guidelines for HW and Software)

I would like to focus more time to something like this, but am very new to Open Source development. Thanks for asking the question here. I agree that this new GUI should be accessible from the start as it is much easier than updating the code later and much more likely to get done.

**Reply by mouxdesign** [reply]
Really like your input and ideas on this topic Jason. I agree with @GBKS that the new Gui for Bitcoin core would be a great one to start to consider accessibility from the start. In terms of presenting the accessibility in a easy to understand way, I found this which is not specific to the bitcoin industry, but if we could create something like this specifically to wallet developers that would be great.

https://github.com/UKHomeOffice/posters/blob/master/accessibility/dos-donts/posters_en-UK/accessibility-posters-set.pdf

Would it be an idea to start with an audit of the Bitcoin core gui as a practical task to begin with? And then creating a resource from there.

Any tools or specific audit points/checklist which we can follow when doing the app audit would be much appreciated!

**Reply by GBKS** [reply]
@jason-me Mo and I have started doing some mobile accessibility testing. Looks like there are plenty of low-hanging fruit (labels, order). One thing we're not sure of is the accessibility of QR codes. Do you know how to best handle them? Can they even be made usable for people with visual impairments?

**Reply by jason-me** [reply]
Yeah so, for Qt I am not 100% sure but I would start with WCAG 2 ICT as a set of guidelines to build a checklist from since the W3Cs Web Content Accessibility Guidelines are created for web not GUI applications, but this document serves as a bridge for non-web appications and content: https://www.w3.org/TR/wcag2ict/

As always testing with mutliple screen readers such as JAWS, NVDA, and VoiceOver will also need to be part of any accessibility test plan along with some screen magnification software such as ZoomText and leveraged accessibility settings within each respective OS (e.g. zoom, high contrast mode, etc). If the components that make up the GUI are accessible and actionable via keyboard and they announce an accessible name, role, and value/state (as needed for drop down menus, etc) when using a screen reader then that usually covers the most impactful accessibility issues in most applications aside from color contrast and readability.

Will look for a good GUI checklist with step by step guidance, but may be best just to create a checklist from WCAG2ICT from scratch.

Also, just browsing the web I found this which may be useful for Qt in particular: https://www.qt.io/blog/tag/accessibility

**Reply by jason-me** [reply]
> @jason-me Mo and I have started doing some mobile accessibility testing. Looks like there are plenty of low-hanging fruit (labels, order). One thing we're not sure of is the accessibility of QR codes. Do you know how to best handle them? Can they even be made usable for people with visual impairments?

For QR codes, I would say that having the ability to copy/paste plain text via keyboard (or gestures on mobile) and also providing a simple text alternative that announces the presence of a QR code and its purpose is the most practical way to make them more accessible, but there may be some QR code applications that provide audible feedback when a QR code is in view (not 100% sure about this but likely since there are some text scanner apps that already do something similar). I also liked the idea someone else had here about NFC being more accessible than a QR code. You can provide a tactile surface that a blind user with a cell phone can feel and identify then use NFC to tap their phone or card as one modality that may be more accessible in those use cases.

**Reply by jason-me** [reply]
Here is the accessibility API documentation for Qt: https://wiki.qt.io/Qt_Accessibility Might be kinda important.
